workitem/link/link_repository.go

```diff
@@ -97,6 +97,30 @@ func (r *GormWorkItemLinkRepository) ValidateCorrectSourceAndTargetType(ctx cont
 	return nil
 }
 
+// CheckParentExist returns error if there is an attempt to create more than 1 parent of a workitem.
+func (r *GormWorkItemLinkRepository) CheckParentExist(ctx context.Context, sourceID, targetID uint64, linkTypeID uuid.UUID) error {
+	const treeTopology = "tree"
+
+	// Fetch the link type
+	linkType, err := r.workItemLinkTypeRepo.LoadTypeFromDBByID(ctx, linkTypeID)
+	if err != nil {
+		return errs.WithStack(err)
+	}
+
+	if linkType.Topology == treeTopology {
+		result := WorkItemLink{}
+
+		// check if the same link type already exists for given target ID
+		db := r.db.Where("link_type_id=? AND target_id=?", linkTypeID, targetID).Find(&result)
+		if db.RecordNotFound() {
+			// not treating this as an error
+			return nil
+		}
+		return errors.NewInternalError(db.Error.Error())
```

Comment:
I'm thinking if an errors.NewBadParameter error is best raised here. Make sure you use `log.Error` as it is done in other (newer) places of the code to log any error from the database before returning an error.